### DISCUSSION TITLE
[Feature] Implementing a Utils class directly in the library

### DISCUSSION THREAD

**Discussion by valentinhttr** [inisos/bank4j#10] [Ideas]
Hi @pblanchardie,

Let me put forward an idea, and I'd like to know what you think. The idea is to offer the most comprehensive library possible, while keeping it light and uncomplicated.

I was wondering if it wouldn't be possible to offer an [Utils Class](https://github.com/dvbern/iso20022/blob/develop/iso20022-ch-impl/src/main/java/ch/dvbern/oss/lib/iso20022/Iso20022Util.java#L202-L226) to help us format certain fields, which could of course be totally optional. Optional, so that we retain control over field formatting and know what we're actually generating.

For example only a restricted set of characters is permitted, the "SWIFT characters set", for the following reference elements:
<img width="851" alt="CleanShot 2023-10-31 at 09 22 23@2x" src="https://github.com/inisos/bank4j/assets/20521112/329c2d9d-e9fb-4697-b2ca-880991137dc0">

Take a look at the page 16 & 17 of the business specifications :
[ig-credit-transfer-v2.0.2-en.pdf](https://github.com/inisos/bank4j/files/13213814/ig-credit-transfer-v2.0.2-en.pdf)

What do you think of the idea of implementing this directly at lib level?

**Comment by valentinhttr** [comment]
@pblanchardie Any update pls?

**Reply by pblanchardie** [reply]
Yes I'll add this in a few days, quite busy at the moment. Thanks for the idea and reference.

Btw, I think that we should also validate the content when generating XML message (or sooner, when building the model) and throw an exception if invalid characters are found.

**Comment by pblanchardie** [comment]
Hi @valh1996, what was the purpose of `allowDiacritics`?

**Reply by valentinhttr** [reply]
I thought I removed it everywhere, but I forgot [here](https://github.com/inisos/bank4j/blob/main/src/main/java/io/inisos/bank4j/validator/constraints/Iso20022CharacterSet.java#L23).
Can you remove it pls? ISO20022 doesn't support diacritics
